app/config/ApplicationConfig.scala

```diff
@@ -16,45 +16,30 @@
-}
-
-object ApplicationConfig extends ApplicationConfig with ServicesConfig with LegacyI18nSupport {
-
-  override protected def mode: Mode = Play.current.mode
-  override protected def runModeConfiguration: Configuration = Play.current.configuration
-
-  private def loadConfig(key: String) = configuration.getString(key).getOrElse(throw new Exception(Messages("ers.exceptions.applicationConfig.missingKey", key)))
-
-  private val contactHost = configuration.getString("contact-frontend.host").getOrElse("")
-
-  override lazy val assetsPrefix: String = loadConfig("assets.url") + loadConfig("assets.version")
-  override lazy val betaFeedbackUrl = s"$contactHost/contact/beta-feedback"
-  override lazy val betaFeedbackUnauthenticatedUrl = s"$contactHost/contact/beta-feedback-unauthenticated"
-  override lazy val analyticsToken: Option[String] = configuration.getString("google-analytics.token")
-  override lazy val analyticsHost: String = configuration.getString("google-analytics.host").getOrElse("auto")
-
-  override lazy val submissionsUrl: String = baseUrl("ers-submissions") + "/ers-submissions/" + "submit-presubmission"
-
-  override lazy val splitLargeSchemes: Boolean = configuration.getBoolean("largefiles.enabled").getOrElse(false)
-  override lazy val maxNumberOfRowsPerSubmission: Int = configuration.getInt("largefiles.maxrowspersheet").getOrElse(10000)
+import javax.inject.{Inject, Singleton}
+import uk.gov.hmrc.play.bootstrap.config.ServicesConfig
+
+import scala.concurrent.duration.{DurationInt, FiniteDuration}
+
+@Singleton
+class ApplicationConfig @Inject()(config: ServicesConfig) {
+
+  private def loadConfig(key: String) = config.getString(key)
+  private val contactHost = config.getConfString("contact-frontend.host", "")
+
+  lazy val assetsPrefix: String = loadConfig("assets.url") + loadConfig("assets.version")
+  lazy val betaFeedbackUrl = s"$contactHost/contact/beta-feedback"
+  lazy val betaFeedbackUnauthenticatedUrl = s"$contactHost/contact/beta-feedback-unauthenticated"
+  lazy val analyticsToken: String = loadConfig("google-analytics.token")
+  lazy val analyticsHost: String = config.getConfString("google-analytics.host", "auto")
+  lazy val fileValidatorBaseUrl: String = config.baseUrl("ers-file-validator")
+  lazy val maxNumberOfRowsPerSubmission: Int = config.getConfInt("largefiles.maxrowspersheet", 10000)
+  lazy val validationChunkSize: Int = config.getConfInt("validationChunkSize", 10000)
+  lazy val submissionsUrl: String = config.baseUrl("ers-submissions")
+  lazy val splitLargeSchemes: Boolean = config.getConfBool("largefiles.enabled", false)
+  lazy val sessionCacheBaseUri: String = config.baseUrl("cachable.session-cache")
+  lazy val ersTimeOut: FiniteDuration = config.getConfInt("ers-file-validator-timeout-seconds", 20).seconds
+  lazy val sessionCacheDomain: String = config.getConfString("cachable.session-cache.domain",
+    throw new Exception("Could not find config ''cachable.session-cache.domain''"))
```

Comment:
Need to be careful here, you've made a change here that means you'll likely only be using default values e.g.
`configuration.getString("google-analytics.host").getOrElse("auto")`**has been updated to**
`config.getConfString("google-analytics.host", "auto")`

However the .getConfString function will append 'microservice.services' to whatever string you pass in so what you've actually done is:
`configuration.getString("microservice.services.google-analytics.host").getOrElse("auto")`
Which, unless you've moved all the config into a new location, won't be found.